website_compassion/controllers/my_account.py

```diff
@@ -269,6 +269,33 @@ def my_child(self, state="active", child_id=None, **kwargs):
             # No child is selected, we pick the first one by default
             return request.redirect(f"/my/children?child_id={children[0].id}")
 
+    @route("/my/donations/update", type="http", auth="user", website=True)
+    def my_donations_update(self, recurring_contract=None, res=None, **kw):
+        partner = request.env.user.partner_id
+        contract = (partner.contracts_fully_managed
+                    + partner.contracts_correspondant) \
+            .filtered(lambda a: a.state not in ["cancelled", "terminated"]).filtered(
+            lambda a: int(a.id) == int(recurring_contract))
```

Comment:
Potential error when `recurring_contract` is not specified. Add redirection if None ?
a.id is already an int